src/server/pkg/storage/chunk/metadata.go

```diff
@@ -0,0 +1,89 @@
+package chunk
+
+import (
+	"context"
+	"crypto/sha512"
+	"encoding/hex"
+
+	"github.com/jmoiron/sqlx"
+)
+
+type ID []byte
+
+func Hash(data []byte) ID {
+	h := sha512.New()
+	h.Write(data)
+	return h.Sum(nil)[:32]
+}
+
+func ChunkIDFromHex(h string) (ID, error) {
+	return hex.DecodeString(h)
+}
+
+func (id ID) HexString() string {
+	return hex.EncodeToString(id)
+}
+
+type Metadata struct {
+	Size     int
+	PointsTo []ID
+}
+
+type MetadataStore interface {
+	// SetChunkInfo adds chunk metadata to the tracker
+	Set(ctx context.Context, chunkID ID, md Metadata) error
+	// GetChunkInfo returns info about the chunk if it exists
+	Get(ctx context.Context, chunkID ID) (*Metadata, error)
+	// DeleteChunkInfo removes chunk metadata from the tracker
+	Delete(ctx context.Context, chunkID ID) error
+}
+
+var _ MetadataStore = &PostgresStore{}
+
+type PostgresStore struct {
+	db *sqlx.DB
+}
+
+func NewPostgresStore(db *sqlx.DB) *PostgresStore {
+	return &PostgresStore{db: db}
+}
+
+func (s *PostgresStore) Set(ctx context.Context, chunkID ID, md Metadata) error {
+	_, err := s.db.ExecContext(ctx,
+		`INSERT INTO storage.chunks (hash_id, size) VALUES ($1, $2)
+		ON CONFLICT (hash_id) DO UPDATE SET size = $2 WHERE storage.chunks.hash_id = $1
```

Comment:
When would the size of a chunk change?

Reply:
It wouldn't but since the method is called `Set`, It should have set/put semantics.

Reply:
We should probably error if there is a conflict. I think we only want to set the hash and size once, and potentially update other fields (such as last accessed for archiving purposes). We probably need to draw a distinction between the queries that set the immutable fields and the queries that set the mutable fields.